pkg/service/annotation.go

```diff
@@ -30,11 +31,17 @@ func (svc AnnotationsService) CreateAnnotation(ctx context.Context, params model
 	tx := svc.db.WithContext(ctx)
 
 	// Upsert
-	if tx.Where(model.CreateAnnotation{
-		AppName:   params.AppName,
-		Timestamp: params.Timestamp,
-	}).Updates(&u).RowsAffected == 0 {
-		return &u, tx.Create(&u).Error
+	if err := tx.Clauses(clause.OnConflict{
+		Columns: []clause.Column{
+			{Name: "app_name"},
+			{Name: "timestamp"},
+		},
+		// Update fields we care about
+		DoUpdates: clause.AssignmentColumns([]string{"app_name", "timestamp", "content"}),
+		// Update updateAt fields
+		UpdateAll: true,
+	}).Create(&u).Error; err != nil {
+		return nil, err
 	}
```

Comment:
Not sure this is actually better.

There are a few things happening here:
1. We need to tell which fields we are interested in conflicting (`app_name`, `timestamp`), otherwise it defaults to the primary key. I decided to not use a natural primary key here, since we may want to for eg modify an existing annotation (change `appName` for example)
2. `DoUpdates` since we need to tell which columns we want to update. Otherwise it doesn't do anything.
3. `UpdateAll` to tell it to update `updateAt` field. Here's where it gets confusing: I thought `UpdateAll` would update everything (duh), but it seems to ignore the other fields.

Reply:
I guess we only need to update `content`:
```go
DoUpdates: clause.AssignmentColumns([]string{"content"}),
```